### ISSUE TITLE
Color code different sources in the lineage graph

### ISSUE THREAD

**Issue by PowdyPowPow** [dbt-labs/dbt-docs#26]
Currently all source tables are shown green in the lineage graph.
It would be helpful if the tables from different sources would be shown in a different color.

![image](https://user-images.githubusercontent.com/33448539/57382635-97b4cc00-71ad-11e9-856a-0ff0e1d4753d.png)

**Comment by drewbanin** [comment]
@PowdyPowPow cool idea! I think it might look kind of wonky to color every source differently... maybe that's just something that we need to experiment with. It could be interesting to tone down the color (like use gray instead of green, or something) and then use a differently colored border, for instance.

What do you think?

**Comment by PowdyPowPow** [comment]
@drewbanin i guess it could become very colorful and you would need to choose "easy" colors. But as currently the sources get distributed all over the graph it is already colorful with just the green. Maybe if you could somehow cluster them together it would probably also be easier to use some more discreet common element like colored border or just a different shades of grey? Or maybe you could group the different source tables together and put them in a (colored) box/container if they were all in the same place?

**Comment by yarodmi** [comment]
Hi Team,

I really think this will be a very useful feature. 
Thou, I would suggest providing the ability to apply custom colour coding (rather than just different sources) as there might be different use cases. 

Few examples where I would love to use it :
- quite often I am having multiple database schemas representing different architecture layers.  Would be great to see them(schemas) in different colours so it is clear whats where
- highlight facts and dimensions in different colors
- highlight different materializations (what is a view and what is a table for example). We often start from 'everything as a view' and then deciding what models are useful to materialize. Seeing this in a graph will be very useful.

To simplify the adoption, might be reasonable to pre-define some colours for some tags (config(tags=["dimension"]) or config(tags=["fact"]) config(tags=["base"]) plus the ability to define a custom mapping. 

Thanks in advance!

**Comment by MartinGuindon** [comment]
I agree with @yarodmi, it would be really useful to be able to define/override color rules.

I really like the idea of being able to define background colors _and_ border colors for the nodes. I'd certainly use it to differentiate sources, staging models and fact/dimension tables. I could easily see this as a simple parameter in the source or dbt_project YAML files, eg:

schema.yml
```yaml
sources:
  - name: retently
     docs_background_color: BFBFC0
     docs_border_color: 3346F3
     ..other source config..

  - name: gsheets
    docs_background_color: BFBFC0
    docs_border_color: 267C0B
    ..other source config..
```

dbt_project.yml
```yaml
models:
  my_project:
    marts:
      docs_background_color: 10C3C3

    staging:
      docs_background_color: 4F0BC4
```

I think it would require being able to set background color, text color and border color at the minimum.

**Comment by bcolbert978** [comment]
Reposting here from the dbt Slack #suggestions channel per Drew's advice - another extension of the lineage graph color-coding I'd like to see (not related to sources, but rather the search filter). When I filter down to specific `--models` it would be helpful if the `model` I name in my search were lit up (kind of like things light up purple when you select them). I often use the `+model+` or `@model` searches here pretty frequently and sometime there are too many models loading in to stay grounded around my search.

**Comment by guy-adams** [comment]
+1 for being able to map colours to tags. One additional thought - may people (e.g. in an ingestion,curation,calcuation,consumption architecture) have clearly defined 'layers' or groupings (e.g. staging, intermediate marts, core marts) of models. To be able to force the models into virtual columns (with a title) would be fantastic e.g. 

![image](https://user-images.githubusercontent.com/46963181/67555469-4079c800-f709-11e9-9037-77e67cad47ca.png)

I'm already denoting these layers using model tags.

**Comment by diegocamelo92** [comment]
I also think this could be very valuable, it would help us identify regions of different sources.

**Comment by ptc-sgauglitz** [comment]
Hi -- would like to upvote both of these feature requests (ability to color code, and ability to set columns for designated layers). This would be super useful, including for reviews such as demoed in your own video here: https://www.youtube.com/watch?v=5W6VrnHVkCA&list=PL0QYlrC86xQmPf9QUceFdOarYcv3ETSsz

(also raised in dbt's Slack community)

Thoughts?

**Comment by tnightengale** [comment]
@drewbanin Hey Drew! Hope all is well :) this feature is quite stale but it's something that our team would love to have. Any suggestions on where to start in the code base to make a contribution to this?

Specifically, implementing alignment "zones" based on tags like in @guy-adams drawing above!

**Comment by JC-Lightfold** [comment]
@drewbanin - columns and colours would be close to my HIGHEST priority right now. Why? Because we're teaching every client how to live/breathe/think DBT and those two things help the DAG be the central unifying concept of the whole. Without them, any moderate sized DAG becomes fairly useless for understanding as a whole.

**Comment by yu-iskw** [comment]
It would be important to visually distinguish nodes by layers. At the same time, I would like to cluster nodes by schema and database in terms of dbt, because our BigQuery tables exist in different google cloud projects and datasets. That would enable us to easily manage and audit dbt models.

![image](https://user-images.githubusercontent.com/1523515/165656918-b372106c-a0c2-431b-928c-7b56a16576ec.png)

**Comment by jtcohen6** [comment]
In v1.3, a custom `node_color` config will be supported for all models, seeds, snapshots, and tests: https://docs.getdbt.com/reference/resource-configs/docs#custom-node-colors

We chose not to extend it to sources, exposures, and metrics for now. Still, that capability should go a way toward enabling a number of the desires in this thread!

**Comment by saraleon1** [comment]
Great news - thanks Jeremy!!

**Comment by belasobral93** [comment]
+1 I would like to visually represent [Databrick's medallion architecture](https://www.databricks.com/glossary/medallion-architecture)

**Comment by serene-capybara** [comment]
> In v1.3, a custom `node_color` config will be supported for all models, seeds, snapshots, and tests: https://docs.getdbt.com/reference/resource-configs/docs#custom-node-colors
> 
> We chose not to extend it to sources, exposures, and metrics for now. Still, that capability should go a way toward enabling a number of the desires in this thread!

How does one use `node_color` on seed nodes?

**Comment by patriklundberg** [comment]
Would it be possible to change the font color of the node name?

**Comment by dbeatty10** [comment]
@serene-capybara to use node color on seeds, you'd do something like this in your `dbt_project.yml`:
```yaml
seeds:
  +docs:
    node_color: "#cd7f32"
```

Or something like this within a YAML file like `models/_seeds.yml`:
```yaml
seeds:
  - name: my_seed
    docs:
      node_color: purple
```

**Comment by dbeatty10** [comment]
> Would it be possible to change the text color of the node name?

@patriklundberg changing the text color is not possible, but https://github.com/dbt-labs/dbt-docs/issues/408 has some discussion of how to achieve higher contrast between the text and the background color.

**Comment by ngouass** [comment]
Hello @everyone, I came accross this issue one week ago while looking for a solution to improve the lineage view of my project.

After unsucessful searchs, I'm planning to explore some solutions that I could build in an external tool. Before starting any development, I would like to know if some participants are still looking for a solution, if yes please let me know with A VOTE on this comment.

Below are potentital features, I was able to extract based on comments in this issue: 
    -> (7) apply custom color coding
    - highlight database schemas
    - highlight facts and dimensions in different colors
    - highlight different materializations (what is a view and what is a table for example).
    - Apply color based on tags
    -> (46) Group model by layers and Highlight layers 
    -> (5) Search functionality with model highlighted when matching search query

Thanks.

**Comment by ptc-sgauglitz** [comment]
Hi @ngouass 

Color coding is now supported since dbt 1.3 (see @jtcohen6 's Sep 6, 2022 post); I believe that can cover items 1-5 on your list (using either tags or model file/pathnames as intermediary to apply the color coding to as needed). I've used it to indicate the different **materializations** (your item 4).

Personally, I'd still be very interested in options to improve the layouting (in particular, ability to have models assigned to certain columns, to represent the layers -- your item 6, I believe). This could be in the form of a strict (column) assignment or as "hints" to the auto-layouting. Whether I'd be interested in using an additional tool for that would depend on the ease of use/ease of integration in the toolchain.

**Comment by polmonso** [comment]
> +1 for being able to map colours to tags. One additional thought - may people (e.g. in an ingestion,curation,calcuation,consumption architecture) have clearly defined 'layers' or groupings (e.g. staging, intermediate marts, core marts) of models. To be able to force the models into virtual columns (with a title) would be fantastic e.g.
> 
> ![image](https://user-images.githubusercontent.com/46963181/67555469-4079c800-f709-11e9-9037-77e67cad47ca.png)
> 
> I'm already denoting these layers using model tags.

DId you open another issue/feature requests with the layer positioning idea?

**Comment by ptc-sgauglitz** [comment]
> DId you open another issue/feature requests with the layer positioning idea?

I did not open another issue nor am I aware of one; it seems both requests were discussed on this one thread.
At least for my purposes, the "color" aspect of it is addressed since dbt 1.3; I would support opening a separate issue specifically for the layouting/"assign columns" aspect. I think this would still be very helpful. I'd be happy to open that issue, or please copy me on it if somebody else opens it.

**Comment by ptc-sgauglitz** [comment]
thanks @mattyb !
https://github.com/dbt-labs/dbt-docs/issues/438 looks more specific it I see that correctly.
https://github.com/dbt-labs/dbt-docs/issues/330 is, to my understanding, quite similar to (at least some of) the asks/suggestions here, but it hasn't seen any traction and has been closed due to inactivity -- which is curious given that the ask here has repeatedly drawn lots of upvotes/interest 🤔 

I'd be inclined to open a new issue, citing the tickets and above and referencing the interest voiced in this thread too 🤔

**Comment by polmonso** [comment]
we're still interested.

**Comment by rafaelamilagres** [comment]
It would be great to change the green of source nodes

**Comment by dbeatty10** [comment]
@rafaelamilagres as you noted, it is not currently possible to use `node_color` for sources, and they will always appear green.

Our documentation notes the node types that support the [`node_color`](https://docs.getdbt.com/reference/resource-configs/docs) property currently:
> The `docs` field can be used to provide documentation-specific configuration to models.
> It also supports `node_color` for models, seeds, snapshots, and analyses. Other node types are not supported.

If you're seeking more node types to be added, could you [open a feature request](https://github.com/dbt-labs/dbt-docs/issues/new?assignees=&labels=enhancement%2C+triage&projects=&template=feature_request.md&title=)?

**Comment by dbeatty10** [comment]
> we're still interested.

@polmonso and/or @ejgal -- what are you interested in precisely? Just coloring of nodes? Or their layout?

If it's the latter, could you comment on either https://github.com/dbt-labs/dbt-docs/issues/330 and https://github.com/dbt-labs/dbt-docs/issues/438 or open a new issue?

**Comment by ejgal** [comment]
Both. For the color part I would just like to change the color of sources. Will look into the issues you linked for the layout part.

**Comment by patriklundberg** [comment]
I would like to change the text color, from white to black. It would make it easier to read the names of the table.



Sent from Outlook for Android<https://aka.ms/AAb9ysg>

________________________________

**Comment by polmonso** [comment]
@dbeatty10 same as @ejgal, both. But more so the layout, to distinguish clearly between raw - staging/intermediate - datamart layers of the data pipeline. I'll comment on the mentioned issues.

**Comment by katy-sadowski** [comment]
@dbeatty10 @jtcohen6  upvoting request to be able to specify a color for source nodes.  should we hope/assume that this issue will be reopened to add that feature, or should i file a new feature request?  thanks!

my use case is simply to be able to choose the color for source nodes instead of being stuck with green, so i can fully customize the appearance of my DAG 😄

**Comment by eliottaudry** [comment]
Upvoting for this long awaited feature, that would be really cool is we could adjust the different sources colors in our documentation to provide more clarity.
